src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -109,18 +109,18 @@ public Task DeleteAgentAsync(int agentPoolId, int agentId)
         }
     }
 
-    public sealed class MachineGroupAgentConfigProvider : AgentService, IConfigurationProvider
+    public sealed class DeploymentGroupAgentConfigProvider : AgentService, IConfigurationProvider
```

Comment:
IMO, it is okay to call it simply _DeploymentAgent_ instead of _DeploymentGroupAgent_